### DISCUSSION TITLE
Persistente Volumes für docker compose

### DISCUSSION THREAD

**Discussion by synchronierer** [jzakotnik/openlibry#83] [Ideas]
Wenn das mal mit mehr Zeit möglich ist, fände ich das Mappen von persistenten Volumen zwischen Host und Container ganz sinnvoll, damit man Einstellungen, Covers, usw. leichter verändern und sichern kann.

Momentan gelingt mir das nur, wenn ich mit /bin/sh in den Container springe und dort die Änderungen vornehme. Käme aber ein neues Image raus und der Container müsste neu aufgesetzt werden, fange ich ja sonst wieder bei null an, oder?

**Comment by jzakotnik** [comment]
Ja, das kann ich relativ leicht im compose File anpassen. Bin großer Docker Fan, aber in dem Fall ist das Aufsetzen einer Node-JS Umgebug mit automatisierten Restart via [pm2](https://pm2.keymetrics.io/) eben doch viel einfacher..

**Reply by synchronierer** [reply]
Ich habe mit Node null Erfahrung und ebensowenig mit pm2.
Ich hoffe, dir da auch nicht zu viel Arbeit zu machen.

Im Fall unserer Schule setze ich auch deswegen auf Container, weil wir zwei getrennte Büchereien haben (Unter-/Mittelstufe und Oberstufe).

**Reply by jzakotnik** [reply]
Im Gegenteil, habe bisher viel fachliches Feedback bekommen aber noch keins zur Installation und Verbesserung der technischen Dokumentation. Danke für die Geduld :-)

**Reply by jzakotnik** [reply]
@synchronierer aktuell wird im [Docker Compose](https://github.com/jzakotnik/openlibry/blob/main/docker-compose.yml) schon die Datenbank auf ein persistent volume gemappt, allerdings die cover noch nicht, das ist richtig..

**Reply by jzakotnik** [reply]
Das docker compose file ist jetzt so angepasst, dass auch die cover images im persistent volume sind. Nicht vergessen den Pfad im .env File entsprechend zu setzen (im `.env_example` bereits korrekt)

**Reply by synchronierer** [reply]
Kann es sein, dass der richige Pfad für dev.db nicht 

:/app/prisma/database

sondern
:/app/prisma/

ist?
